### PR TITLE
Logging improvements

### PR DESCRIPTION
**Pull request by kgaillot**
Reword various messages for clarity and consistency, drop some redundant messages, and refactor some functions to return standard return codes so error messages can give sufficient detail.

### CODE REVIEW

**Comment by kgaillot** [comment]
@clumens , can you review this one too? Thanks

File: lib/pacemaker/pcmk_sched_utils.c [hunk 1]
```diff
@@ -787,3 +787,59 @@ pcmk__new_logger(void)
...
+ * \param[in] node      Node to check
+ * \param[in] data_set  Cluster working set
+ *
+ * \return Failed resource (possibly a parent of \p rsc) if migration threshold
+ *         has been reached, or NULL otherwise
+ */
+pe_resource_t *
+pcmk__threshold_reached(pe_resource_t *rsc, pe_node_t *node,
+                        pe_working_set_t *data_set)
+{
```
Comment (clumens) [inline lib/pacemaker/pcmk_sched_utils.c:804]: The name seems weird to me.  I'd expect pcmk__threshold_reached to return a bool like the old function did.  Maybe something like pcmk__rsc_over_threshold?  That kind of seems bool-like to me too.

File: lib/cluster/cpg.c [hunk 2]
```diff
@@ -252,13 +252,10 @@ crm_cs_flush(gpointer data)
...
 
     queue_len -= sent;
-    if ((sent > 1) || (cs_message_queue != NULL)) {
-        crm_info("Sent %u CPG messages  (%d remaining): %s (%d)",
-                 sent, queue_len, pcmk__cs_err_str(rc), (int) rc);
-    } else {
-        crm_trace("Sent %u CPG messages  (%d remaining): %s (%d)",
-                  sent, queue_len, pcmk__cs_err_str(rc), (int) rc);
-    }
+    do_crm_log((queue_len > 5)? LOG_INFO : LOG_TRACE,
```
Comment (clumens) [inline lib/cluster/cpg.c:255]: I see at least one other place where we use 5 regarding queue_len.  Should this be a constant somewhere?

Comment (wenningerk) [reply lib/cluster/cpg.c:255]: or LOG_PRIO_TRACE or something ...

File: lib/pacemaker/pcmk_sched_utils.c (hunk 1 above)
Comment (kgaillot) [reply lib/pacemaker/pcmk_sched_utils.c:804]: I could make it return bool, and take the failed resource as an output argument

File: lib/cluster/cpg.c (hunk 2 above)
Comment (kgaillot) [reply lib/cluster/cpg.c:255]: They're only marginally related; this is the CPG message queue and the other use (presumably delay_next_flush()) is the IPC message queue. This is purely a log message, whereas the other affects the sending of messages.

The use here isn't so much a magic number as a ballpark figure for when to complain more loudly.

File: lib/pacemaker/pcmk_sched_utils.c (hunk 1 above)
Comment (kgaillot) [reply lib/pacemaker/pcmk_sched_utils.c:805]: Added commit on top for this

Comment (clumens) [reply lib/pacemaker/pcmk_sched_utils.c:805]: That works for me.

**Pull request merged**